src/python/grpcio_tests/tests_aio/unit/server_test.py

```diff
@@ -458,6 +470,29 @@ async def test_error_without_raise_in_unary_unary(self):
         rpc_error = exception_context.exception
         self.assertEqual(grpc.StatusCode.INTERNAL, rpc_error.code())
 
+    async def test_error_without_raise_in_stream_unary(self):
+
+        async def request_iterator():
+            for _ in range(_NUM_STREAM_REQUESTS):
+                yield _REQUEST
+
+        call = self._channel.stream_unary(
+            _ERROR_WITHOUT_RAISE_IN_STREAM_UNARY)(request_iterator())
+
+        self.assertEqual(grpc.StatusCode.INTERNAL, await call.code())
+        self.assertFalse(call.cancelled())
+        self.assertTrue(call._async_request_poller.done())
+
+        call = self._channel.stream_unary(
+            _ERROR_WITHOUT_RAISE_IN_STREAM_UNARY)(request_iterator())
+
+        with self.assertRaises(aio.AioRpcError) as exception_context:
+            await call
+        self.assertEqual(grpc.StatusCode.INTERNAL,
+                         exception_context.exception.code())
+        self.assertFalse(call.cancelled())
+        self.assertTrue(call._async_request_poller.done())
```

Comment:
What I'd really like to assert is that after my change you don't get THIS:

```
$ pytest src/python/grpcio_tests/tests_aio/unit/server_test.py::TestServer::test_error_without_raise_in_stream_unary
=========================================================================================== test session starts ===========================================================================================
platform linux -- Python 3.8.5, pytest-6.2.2, py-1.10.0, pluggy-0.13.1
rootdir: /home/clayg/Workspace/scratch/grpc-test/.scratch/grpc/src/python/grpcio_tests
collected 1 item                                                                                                                                                                                          

src/python/grpcio_tests/tests_aio/unit/server_test.py .                                                                                                                                             [100%]

============================================================================================ warnings summary =============================================================================================
tests_aio/unit/server_test.py::TestServer::test_error_without_raise_in_stream_unary
  /home/clayg/Workspace/scratch/grpc-test/.scratch/grpc/src/python/grpcio/grpc/aio/_server.py:53: DeprecationWarning: The loop argument is deprecated since Python 3.8, and scheduled for removal in Python 3.10.
    self._server = cygrpc.AioServer(

tests_aio/unit/server_test.py::TestServer::test_error_without_raise_in_stream_unary
tests_aio/unit/server_test.py::TestServer::test_error_without_raise_in_stream_unary
  /home/clayg/Workspace/scratch/grpc-test/.scratch/grpc/src/python/grpcio/grpc/aio/_call.py:369: DeprecationWarning: The loop argument is deprecated since Python 3.8, and scheduled for removal in Python 3.10.
    self._metadata_sent = asyncio.Event(loop=self._loop)

-- Docs: https://docs.pytest.org/en/stable/warnings.html
====================================================================================== 1 passed, 3 warnings in 0.12s ======================================================================================
Task exception was never retrieved
future: <Task finished name='Task-8' coro=<_StreamRequestMixin._consume_request_iterator() done, defined at /home/clayg/Workspace/scratch/grpc-test/.scratch/grpc/src/python/grpcio/grpc/aio/_call.py:396> exception=ExecuteBatchError('Failed "execute_batch": (<grpc._cython.cygrpc.SendMessageOperation object at 0x7fe678ac59b0>,)') created at /home/clayg/Workspace/scratch/grpc-test/.scratch/grpc/src/python/grpcio/grpc/aio/_call.py:374>
source_traceback: Object created at (most recent call last):
  File "/home/clayg/.local/share/virtualenvs/grpc-test-ywCqvyqB/bin/pytest", line 8, in <module>
    sys.exit(console_main())
  File "/home/clayg/.local/share/virtualenvs/grpc-test-ywCqvyqB/lib/python3.8/site-packages/_pytest/config/__init__.py", line 185, in console_main
    code = main()
  File "/home/clayg/.local/share/virtualenvs/grpc-test-ywCqvyqB/lib/python3.8/site-packages/_pytest/config/__init__.py", line 162, in main
    ret: Union[ExitCode, int] = config.hook.pytest_cmdline_main(
  File "/home/clayg/.local/share/virtualenvs/grpc-test-ywCqvyqB/lib/python3.8/site-packages/pluggy/hooks.py", line 286, in __call__
    return self._hookexec(self, self.get_hookimpls(), kwargs)
  File "/home/clayg/.local/share/virtualenvs/grpc-test-ywCqvyqB/lib/python3.8/site-packages/pluggy/manager.py", line 93, in _hookexec
    return self._inner_hookexec(hook, methods, kwargs)
  File "/home/clayg/.local/share/virtualenvs/grpc-test-ywCqvyqB/lib/python3.8/site-packages/pluggy/manager.py", line 84, in <lambda>
    self._inner_hookexec = lambda hook, methods, kwargs: hook.multicall(
  File "/home/clayg/.local/share/virtualenvs/grpc-test-ywCqvyqB/lib/python3.8/site-packages/pluggy/callers.py", line 187, in _multicall
    res = hook_impl.function(*args)
  File "/home/clayg/.local/share/virtualenvs/grpc-test-ywCqvyqB/lib/python3.8/site-packages/_pytest/main.py", line 316, in pytest_cmdline_main
    return wrap_session(config, _main)
  File "/home/clayg/.local/share/virtualenvs/grpc-test-ywCqvyqB/lib/python3.8/site-packages/_pytest/main.py", line 269, in wrap_session
    session.exitstatus = doit(config, session) or 0
  File "/home/clayg/.local/share/virtualenvs/grpc-test-ywCqvyqB/lib/python3.8/site-packages/_pytest/main.py", line 323, in _main
    config.hook.pytest_runtestloop(session=session)
  File "/home/clayg/.local/share/virtualenvs/grpc-test-ywCqvyqB/lib/python3.8/site-packages/pluggy/hooks.py", line 286, in __call__
    return self._hookexec(self, self.get_hookimpls(), kwargs)
  File "/home/clayg/.local/share/virtualenvs/grpc-test-ywCqvyqB/lib/python3.8/site-packages/pluggy/manager.py", line 93, in _hookexec
    return self._inner_hookexec(hook, methods, kwargs)
  File "/home/clayg/.local/share/virtualenvs/grpc-test-ywCqvyqB/lib/python3.8/site-packages/pluggy/manager.py", line 84, in <lambda>
    self._inner_hookexec = lambda hook, methods, kwargs: hook.multicall(
  File "/home/clayg/.local/share/virtualenvs/grpc-test-ywCqvyqB/lib/python3.8/site-packages/pluggy/callers.py", line 187, in _multicall
    res = hook_impl.function(*args)
  File "/home/clayg/.local/share/virtualenvs/grpc-test-ywCqvyqB/lib/python3.8/site-packages/_pytest/main.py", line 348, in pytest_runtestloop
    item.config.hook.pytest_runtest_protocol(item=item, nextitem=nextitem)
  File "/home/clayg/.local/share/virtualenvs/grpc-test-ywCqvyqB/lib/python3.8/site-packages/pluggy/hooks.py", line 286, in __call__
    return self._hookexec(self, self.get_hookimpls(), kwargs)
  File "/home/clayg/.local/share/virtualenvs/grpc-test-ywCqvyqB/lib/python3.8/site-packages/pluggy/manager.py", line 93, in _hookexec
    return self._inner_hookexec(hook, methods, kwargs)
  File "/home/clayg/.local/share/virtualenvs/grpc-test-ywCqvyqB/lib/python3.8/site-packages/pluggy/manager.py", line 84, in <lambda>
    self._inner_hookexec = lambda hook, methods, kwargs: hook.multicall(
  File "/home/clayg/.local/share/virtualenvs/grpc-test-ywCqvyqB/lib/python3.8/site-packages/pluggy/callers.py", line 187, in _multicall
    res = hook_impl.function(*args)
  File "/home/clayg/.local/share/virtualenvs/grpc-test-ywCqvyqB/lib/python3.8/site-packages/_pytest/runner.py", line 109, in pytest_runtest_protocol
    runtestprotocol(item, nextitem=nextitem)
  File "/home/clayg/.local/share/virtualenvs/grpc-test-ywCqvyqB/lib/python3.8/site-packages/_pytest/runner.py", line 126, in runtestprotocol
    reports.append(call_and_report(item, "call", log))
  File "/home/clayg/.local/share/virtualenvs/grpc-test-ywCqvyqB/lib/python3.8/site-packages/_pytest/runner.py", line 215, in call_and_report
    call = call_runtest_hook(item, when, **kwds)
  File "/home/clayg/.local/share/virtualenvs/grpc-test-ywCqvyqB/lib/python3.8/site-packages/_pytest/runner.py", line 254, in call_runtest_hook
    return CallInfo.from_call(
  File "/home/clayg/.local/share/virtualenvs/grpc-test-ywCqvyqB/lib/python3.8/site-packages/_pytest/runner.py", line 311, in from_call
    result: Optional[TResult] = func()
  File "/home/clayg/.local/share/virtualenvs/grpc-test-ywCqvyqB/lib/python3.8/site-packages/_pytest/runner.py", line 255, in <lambda>
    lambda: ihook(item=item, **kwds), when=when, reraise=reraise
  File "/home/clayg/.local/share/virtualenvs/grpc-test-ywCqvyqB/lib/python3.8/site-packages/pluggy/hooks.py", line 286, in __call__
    return self._hookexec(self, self.get_hookimpls(), kwargs)
  File "/home/clayg/.local/share/virtualenvs/grpc-test-ywCqvyqB/lib/python3.8/site-packages/pluggy/manager.py", line 93, in _hookexec
    return self._inner_hookexec(hook, methods, kwargs)
  File "/home/clayg/.local/share/virtualenvs/grpc-test-ywCqvyqB/lib/python3.8/site-packages/pluggy/manager.py", line 84, in <lambda>
    self._inner_hookexec = lambda hook, methods, kwargs: hook.multicall(
  File "/home/clayg/.local/share/virtualenvs/grpc-test-ywCqvyqB/lib/python3.8/site-packages/pluggy/callers.py", line 187, in _multicall
    res = hook_impl.function(*args)
  File "/home/clayg/.local/share/virtualenvs/grpc-test-ywCqvyqB/lib/python3.8/site-packages/_pytest/runner.py", line 162, in pytest_runtest_call
    item.runtest()
  File "/home/clayg/.local/share/virtualenvs/grpc-test-ywCqvyqB/lib/python3.8/site-packages/_pytest/unittest.py", line 321, in runtest
    self._testcase(result=self)  # type: ignore[arg-type]
  File "/usr/lib/python3.8/unittest/case.py", line 736, in __call__
    return self.run(*args, **kwds)
  File "/usr/lib/python3.8/unittest/case.py", line 676, in run
    self._callTestMethod(testMethod)
  File "/usr/lib/python3.8/unittest/case.py", line 633, in _callTestMethod
    method()
  File "/home/clayg/Workspace/scratch/grpc-test/.scratch/grpc/src/python/grpcio_tests/tests_aio/unit/_test_base.py", line 31, in wrapper
    return loop.run_until_complete(f(*args, **kwargs))
  File "/usr/lib/python3.8/asyncio/base_events.py", line 603, in run_until_complete
    self.run_forever()
  File "/usr/lib/python3.8/asyncio/base_events.py", line 570, in run_forever
    self._run_once()
  File "/usr/lib/python3.8/asyncio/base_events.py", line 1851, in _run_once
    handle._run()
  File "/usr/lib/python3.8/asyncio/events.py", line 81, in _run
    self._context.run(self._callback, *self._args)
  File "/home/clayg/Workspace/scratch/grpc-test/.scratch/grpc/src/python/grpcio_tests/tests_aio/unit/server_test.py", line 486, in test_error_without_raise_in_stream_unary
    call = self._channel.stream_unary(
  File "/home/clayg/Workspace/scratch/grpc-test/.scratch/grpc/src/python/grpcio/grpc/aio/_channel.py", line 182, in __call__
    call = StreamUnaryCall(request_iterator, deadline, metadata,
  File "/home/clayg/Workspace/scratch/grpc-test/.scratch/grpc/src/python/grpcio/grpc/aio/_call.py", line 576, in __init__
    self._init_stream_request_mixin(request_iterator)
  File "/home/clayg/Workspace/scratch/grpc-test/.scratch/grpc/src/python/grpcio/grpc/aio/_call.py", line 374, in _init_stream_request_mixin
    self._async_request_poller = self._loop.create_task(
Traceback (most recent call last):
  File "/home/clayg/Workspace/scratch/grpc-test/.scratch/grpc/src/python/grpcio/grpc/aio/_call.py", line 403, in _consume_request_iterator
    await self._write(request)
  File "/home/clayg/Workspace/scratch/grpc-test/.scratch/grpc/src/python/grpcio/grpc/aio/_call.py", line 429, in _write
    await self._cython_call.send_serialized_message(serialized_request)
  File "src/python/grpcio/grpc/_cython/_cygrpc/aio/call.pyx.pxi", line 370, in send_serialized_message
    await _send_message(self,
  File "src/python/grpcio/grpc/_cython/_cygrpc/aio/callback_common.pyx.pxi", line 147, in _send_message
    await execute_batch(grpc_call_wrapper, ops, loop)
  File "src/python/grpcio/grpc/_cython/_cygrpc/aio/callback_common.pyx.pxi", line 98, in execute_batch
    await future
grpc._cython.cygrpc.ExecuteBatchError: Failed "execute_batch": (<grpc._cython.cygrpc.SendMessageOperation object at 0x7fe678ac59b0>,)
Task exception was never retrieved
future: <Task finished name='Task-4' coro=<_StreamRequestMixin._consume_request_iterator() done, defined at /home/clayg/Workspace/scratch/grpc-test/.scratch/grpc/src/python/grpcio/grpc/aio/_call.py:396> exception=ExecuteBatchError('Failed "execute_batch": (<grpc._cython.cygrpc.SendMessageOperation object at 0x7fe678ac5d30>,)') created at /home/clayg/Workspace/scratch/grpc-test/.scratch/grpc/src/python/grpcio/grpc/aio/_call.py:374>
source_traceback: Object created at (most recent call last):
  File "/home/clayg/.local/share/virtualenvs/grpc-test-ywCqvyqB/bin/pytest", line 8, in <module>
    sys.exit(console_main())
  File "/home/clayg/.local/share/virtualenvs/grpc-test-ywCqvyqB/lib/python3.8/site-packages/_pytest/config/__init__.py", line 185, in console_main
    code = main()
  File "/home/clayg/.local/share/virtualenvs/grpc-test-ywCqvyqB/lib/python3.8/site-packages/_pytest/config/__init__.py", line 162, in main
    ret: Union[ExitCode, int] = config.hook.pytest_cmdline_main(
  File "/home/clayg/.local/share/virtualenvs/grpc-test-ywCqvyqB/lib/python3.8/site-packages/pluggy/hooks.py", line 286, in __call__
    return self._hookexec(self, self.get_hookimpls(), kwargs)
  File "/home/clayg/.local/share/virtualenvs/grpc-test-ywCqvyqB/lib/python3.8/site-packages/pluggy/manager.py", line 93, in _hookexec
    return self._inner_hookexec(hook, methods, kwargs)
  File "/home/clayg/.local/share/virtualenvs/grpc-test-ywCqvyqB/lib/python3.8/site-packages/pluggy/manager.py", line 84, in <lambda>
    self._inner_hookexec = lambda hook, methods, kwargs: hook.multicall(
  File "/home/clayg/.local/share/virtualenvs/grpc-test-ywCqvyqB/lib/python3.8/site-packages/pluggy/callers.py", line 187, in _multicall
    res = hook_impl.function(*args)
  File "/home/clayg/.local/share/virtualenvs/grpc-test-ywCqvyqB/lib/python3.8/site-packages/_pytest/main.py", line 316, in pytest_cmdline_main
    return wrap_session(config, _main)
  File "/home/clayg/.local/share/virtualenvs/grpc-test-ywCqvyqB/lib/python3.8/site-packages/_pytest/main.py", line 269, in wrap_session
    session.exitstatus = doit(config, session) or 0
  File "/home/clayg/.local/share/virtualenvs/grpc-test-ywCqvyqB/lib/python3.8/site-packages/_pytest/main.py", line 323, in _main
    config.hook.pytest_runtestloop(session=session)
  File "/home/clayg/.local/share/virtualenvs/grpc-test-ywCqvyqB/lib/python3.8/site-packages/pluggy/hooks.py", line 286, in __call__
    return self._hookexec(self, self.get_hookimpls(), kwargs)
  File "/home/clayg/.local/share/virtualenvs/grpc-test-ywCqvyqB/lib/python3.8/site-packages/pluggy/manager.py", line 93, in _hookexec
    return self._inner_hookexec(hook, methods, kwargs)
  File "/home/clayg/.local/share/virtualenvs/grpc-test-ywCqvyqB/lib/python3.8/site-packages/pluggy/manager.py", line 84, in <lambda>
    self._inner_hookexec = lambda hook, methods, kwargs: hook.multicall(
  File "/home/clayg/.local/share/virtualenvs/grpc-test-ywCqvyqB/lib/python3.8/site-packages/pluggy/callers.py", line 187, in _multicall
    res = hook_impl.function(*args)
  File "/home/clayg/.local/share/virtualenvs/grpc-test-ywCqvyqB/lib/python3.8/site-packages/_pytest/main.py", line 348, in pytest_runtestloop
    item.config.hook.pytest_runtest_protocol(item=item, nextitem=nextitem)
  File "/home/clayg/.local/share/virtualenvs/grpc-test-ywCqvyqB/lib/python3.8/site-packages/pluggy/hooks.py", line 286, in __call__
    return self._hookexec(self, self.get_hookimpls(), kwargs)
  File "/home/clayg/.local/share/virtualenvs/grpc-test-ywCqvyqB/lib/python3.8/site-packages/pluggy/manager.py", line 93, in _hookexec
    return self._inner_hookexec(hook, methods, kwargs)
  File "/home/clayg/.local/share/virtualenvs/grpc-test-ywCqvyqB/lib/python3.8/site-packages/pluggy/manager.py", line 84, in <lambda>
    self._inner_hookexec = lambda hook, methods, kwargs: hook.multicall(
  File "/home/clayg/.local/share/virtualenvs/grpc-test-ywCqvyqB/lib/python3.8/site-packages/pluggy/callers.py", line 187, in _multicall
    res = hook_impl.function(*args)
  File "/home/clayg/.local/share/virtualenvs/grpc-test-ywCqvyqB/lib/python3.8/site-packages/_pytest/runner.py", line 109, in pytest_runtest_protocol
    runtestprotocol(item, nextitem=nextitem)
  File "/home/clayg/.local/share/virtualenvs/grpc-test-ywCqvyqB/lib/python3.8/site-packages/_pytest/runner.py", line 126, in runtestprotocol
    reports.append(call_and_report(item, "call", log))
  File "/home/clayg/.local/share/virtualenvs/grpc-test-ywCqvyqB/lib/python3.8/site-packages/_pytest/runner.py", line 215, in call_and_report
    call = call_runtest_hook(item, when, **kwds)
  File "/home/clayg/.local/share/virtualenvs/grpc-test-ywCqvyqB/lib/python3.8/site-packages/_pytest/runner.py", line 254, in call_runtest_hook
    return CallInfo.from_call(
  File "/home/clayg/.local/share/virtualenvs/grpc-test-ywCqvyqB/lib/python3.8/site-packages/_pytest/runner.py", line 311, in from_call
    result: Optional[TResult] = func()
  File "/home/clayg/.local/share/virtualenvs/grpc-test-ywCqvyqB/lib/python3.8/site-packages/_pytest/runner.py", line 255, in <lambda>
    lambda: ihook(item=item, **kwds), when=when, reraise=reraise
  File "/home/clayg/.local/share/virtualenvs/grpc-test-ywCqvyqB/lib/python3.8/site-packages/pluggy/hooks.py", line 286, in __call__
    return self._hookexec(self, self.get_hookimpls(), kwargs)
  File "/home/clayg/.local/share/virtualenvs/grpc-test-ywCqvyqB/lib/python3.8/site-packages/pluggy/manager.py", line 93, in _hookexec
    return self._inner_hookexec(hook, methods, kwargs)
  File "/home/clayg/.local/share/virtualenvs/grpc-test-ywCqvyqB/lib/python3.8/site-packages/pluggy/manager.py", line 84, in <lambda>
    self._inner_hookexec = lambda hook, methods, kwargs: hook.multicall(
  File "/home/clayg/.local/share/virtualenvs/grpc-test-ywCqvyqB/lib/python3.8/site-packages/pluggy/callers.py", line 187, in _multicall
    res = hook_impl.function(*args)
  File "/home/clayg/.local/share/virtualenvs/grpc-test-ywCqvyqB/lib/python3.8/site-packages/_pytest/runner.py", line 162, in pytest_runtest_call
    item.runtest()
  File "/home/clayg/.local/share/virtualenvs/grpc-test-ywCqvyqB/lib/python3.8/site-packages/_pytest/unittest.py", line 321, in runtest
    self._testcase(result=self)  # type: ignore[arg-type]
  File "/usr/lib/python3.8/unittest/case.py", line 736, in __call__
    return self.run(*args, **kwds)
  File "/usr/lib/python3.8/unittest/case.py", line 676, in run
    self._callTestMethod(testMethod)
  File "/usr/lib/python3.8/unittest/case.py", line 633, in _callTestMethod
    method()
  File "/home/clayg/Workspace/scratch/grpc-test/.scratch/grpc/src/python/grpcio_tests/tests_aio/unit/_test_base.py", line 31, in wrapper
    return loop.run_until_complete(f(*args, **kwargs))
  File "/usr/lib/python3.8/asyncio/base_events.py", line 603, in run_until_complete
    self.run_forever()
  File "/usr/lib/python3.8/asyncio/base_events.py", line 570, in run_forever
    self._run_once()
  File "/usr/lib/python3.8/asyncio/base_events.py", line 1851, in _run_once
    handle._run()
  File "/usr/lib/python3.8/asyncio/events.py", line 81, in _run
    self._context.run(self._callback, *self._args)
  File "/home/clayg/Workspace/scratch/grpc-test/.scratch/grpc/src/python/grpcio_tests/tests_aio/unit/server_test.py", line 479, in test_error_without_raise_in_stream_unary
    call = self._channel.stream_unary(
  File "/home/clayg/Workspace/scratch/grpc-test/.scratch/grpc/src/python/grpcio/grpc/aio/_channel.py", line 182, in __call__
    call = StreamUnaryCall(request_iterator, deadline, metadata,
  File "/home/clayg/Workspace/scratch/grpc-test/.scratch/grpc/src/python/grpcio/grpc/aio/_call.py", line 576, in __init__
    self._init_stream_request_mixin(request_iterator)
  File "/home/clayg/Workspace/scratch/grpc-test/.scratch/grpc/src/python/grpcio/grpc/aio/_call.py", line 374, in _init_stream_request_mixin
    self._async_request_poller = self._loop.create_task(
Traceback (most recent call last):
  File "/home/clayg/Workspace/scratch/grpc-test/.scratch/grpc/src/python/grpcio/grpc/aio/_call.py", line 403, in _consume_request_iterator
    await self._write(request)
  File "/home/clayg/Workspace/scratch/grpc-test/.scratch/grpc/src/python/grpcio/grpc/aio/_call.py", line 429, in _write
    await self._cython_call.send_serialized_message(serialized_request)
  File "src/python/grpcio/grpc/_cython/_cygrpc/aio/call.pyx.pxi", line 370, in send_serialized_message
    await _send_message(self,
  File "src/python/grpcio/grpc/_cython/_cygrpc/aio/callback_common.pyx.pxi", line 147, in _send_message
    await execute_batch(grpc_call_wrapper, ops, loop)
  File "src/python/grpcio/grpc/_cython/_cygrpc/aio/callback_common.pyx.pxi", line 98, in execute_batch
    await future
grpc._cython.cygrpc.ExecuteBatchError: Failed "execute_batch": (<grpc._cython.cygrpc.SendMessageOperation object at 0x7fe678ac5d30>,)
```

Reply:
Thanks for taking care of this!